src/main/java/org/spectrumauctions/sats/core/model/mrvm/MRVMRegionalBidder.java

```diff
@@ -85,10 +85,15 @@ public BigDecimal gammaFactor(MRVMRegionsMap.Region r, Bundle<MRVMLicense> bundl
     public Map<MRVMRegionsMap.Region, BigDecimal> gammaFactors(Bundle<MRVMLicense> bundle) {
         Map<MRVMRegionsMap.Region, BigDecimal> result = new HashMap<>();
         for (MRVMRegionsMap.Region region : getWorld().getRegionsMap().getRegions()) {
-            //Note repeately calculating distance is not expensive, as distance is cached in Map Instance
+            // Note that repeatedly calculating distance is not expensive, as distance is cached in Map Instance
             int distance = getWorld().getRegionsMap().getDistance(home, region);
-            BigDecimal discount = distanceDiscounts.get(distance);
-            result.put(region, discount);
+            if (distance > distanceDiscounts.lastKey()) {
```

Comment:
I'm not quite sure about what you're doing here  ;-)
Is your goal to get a map which guarantees a key for every distance, but with a default value for those where the calculation doesn't make sense? If so, have you considered using 'map.getOrDefault(key, defaultvalue)' instead? A pretty neat new function added to Java with version 1.8.
Anyways, the previous code did not look better at all and we can definitely merge as you propose here...

Reply:
As far as I remember, on a random map, there could be non-connected regions, so you're perfectly right about your assumption. And also about the way more elegant way of getOrDefault (which I would know actually, didn't think about it). I'll change it and then merge it. Thanks for the review!